### DISCUSSION TITLE
Do I need to configure MacOS firewall or any other security settings to allow receiving?

### DISCUSSION THREAD

**Discussion by LazyTravellers** [localsend/localsend#1756] [Discussion (English, International)]
Hello All,

I am having issue receiving files / images from android devices to the Mac, however no issues at all receiving to the iPad.  Are there any security settings block in inbound traffic to the Mac?

Thanks

**Comment by Reddyprasadmudiveti** [comment]
No not at all ; 
Try again .

**Reply by adamency** [reply]
stop answering some thread without even verifying by yourself the context of the post. NO. It doesn't work.

**Comment by Mayurifag** [comment]
Having same problem, I am not sure what has to be done in Mac so app will start working and be able to find any other device

Other devices work fine each together

Mac was working previously though (before os update?)

**Reply by Tienisto** [reply]
Do you use a custom download path?

**Reply by Mayurifag** [reply]
@Tienisto no I am not, defaults everywhere in everything 😄 

So the debug info might be helpful, lets say there is MacOS and iOS devices with LocalSend.

I can't find iOS and can't send files with manually typed IP address to iOS from Macbook (it just writes "Error" in interface, how to debug that??)

Though from iOS I can see Macbook and send info — works how it should work! 👍🏻 

I am struggling like hell here and that eats all my emotions, though I know that you are overwhelmed with such issues from other users so it feels disgusting to ask for help but still I need it here 😞 If we will win here in this situation, I will try to add helpful information somewhere in issues/threads/readme

**Reply by Tienisto** [reply]
I agree that "Error" is not helpful. I will make it more explicit in the next release. As a workaround, you can share via link, so that the iOS device downloads the file in the browser.

It would help if you have another device to check if iOS or macOS is the problem.

**Reply by Tienisto** [reply]
I also had this case right now but I don't know why it works again. Basically, I turned off and on "local area" permission in the system settings, then restarted the app several times. Send a message from iPhone to another device, and then the iPhone was discoverable.

I don't know what step exactly made it work. Maybe (1) Resetting the "local area" permission and restarting the app, (2) or sending a file from iPhone to another device

**Reply by Mayurifag** [reply]
Seems a little bit strange for me now. I've tested latest 1.16.1 and just in case I've disabled encryption so it won't bother us here (though, the same thing goes with ssl as well, just didn't want to care about certs for this case)

If it is right error, app tries to go to `http://192.168.69.139:53317/api/localsend/v1/info` in my case. Somehow it struggles to open it. Though, it is easily openable by curl or chrome:

```
{
  "alias": "xxx",
  "version": "2.1",
  "deviceModel": "iPhone",
  "deviceType": "mobile",
  "fingerprint": "xxxx",
  "download": false
}
```

Maybe that is some libraries problem or I dont know..

UPDATE: I thought it might be rust libraries issues, though problem persists on pre-rust localsend versions

<img width="838" alt="image" src="https://github.com/user-attachments/assets/cdb398c3-5060-4953-9625-1a988d8edd2e">

**Reply by Mayurifag** [reply]
HOLY SH*T I SOMEHOW MANAGED IT TO START WORKING

At least MacOS just found the device and sent some text as it had to be.

Here is what I have done:

- Removed everything related to Localsend (via homebrew or Raycast)
- Privacy -> Local Network -> disable Localsend
- Reboot
- Privacy -> Local Network -> there has to be no Localsend entry (!) (that's the main point I guess)
- Install via homebrew as usual, launch. You have to see dialog with Local Network and Allow it!
- ... it didn't work for me here still ... App did not get right of Local Network?
- Reboot
- ?? It starts working somehow ??

I will write in future if it will still work, though that info might be in troubleshooting section — preferably with button to restart app with cleaned Local Network setting. Might be helpful before finding a solution.

**Reply by Tienisto** [reply]
Awesome! Hopefully, we can find the root cause!

**Reply by adamency** [reply]
@Mayurifag What do you mean by `Privacy -> Local Network -> disable Localsend` ? In system settings ? cause there is no "Local Network" section in system settings. Can you please be more precise so we can follow your instructions ? That would be greatly appreciated.

**Reply by tebriel** [reply]
@adamency In system settings (I'm on OSX 15.4.1) there's a `Privacy & Security` panel, inside it is a `Local Network` panel and then you should see a list of applications that have local network access and `LocalSend` will (should?) be one of them. This process didn't work for me so I think Mayurifag may have just gotten lucky (or I did it wrong)!

**Comment by j471n** [comment]
@Tienisto Hi, I am still facing the same problem still. Any Fix?

<img width="832" alt="image" src="https://github.com/user-attachments/assets/76ed267a-5ea8-47aa-9ac3-24c6617ff642" />

**Comment by Jay-Schneider** [comment]
Hello there,
I stumbled upon this thread while googling and just found a solution for me that I'd like to share. I am currently using MacOS Sequoia 15.6.

Aside of the suggestions above
1. Reboot the device after installing via `brew install --cask localsend`
2. Make sure that the App `LocalSend` appears in the "System Settings -> Privacy & Security ->  Local Network" allow list (and is enabled!)

I had to 
- have a look into the firewall settings ("System Settings -> Network -> Firewall -> Options..."). 
- My firewall was configured to disable _all inbound connections_ which is clearly not desirable here. I disabled that. 
- In the configuration list below I saw that there also was already an entry for `LocalSend.app` which was set to explicitly block inbound connections. I set this to allow inbound connections.

After that I was able to not only find the macbook reliably on my phone, I could also fav it without the error message posted above and eventually send files 🥳 

I hope others can profit from that description.
